packages/lesswrong/components/users/UsersSingle.jsx

```diff
@@ -1,10 +1,18 @@
-import { Components, registerComponent } from 'meteor/vulcan:core';
+import { Components, registerComponent, Utils } from 'meteor/vulcan:core';
 import React from 'react';
+import { withRouter } from 'react-router';
 
-const UsersSingle = (props, context) => {
-  return <Components.UsersProfile userId={props.params._id} slug={props.params.slug} />
+const UsersSingle = ({params, router}) => {
+  const slug = Utils.slugify(params.slug);
+  const canonicalUrl = `/users/${slug}`;
+  if (router.location.pathname !== canonicalUrl) {
+    router.replace(canonicalUrl);
```

Comment:
Should we do a navigation event here? Does `router.replace` trigger a navigation event? 

Seems like either way, I would want to add a small comment clarifying whether it does or does not, and I think I prefer us firing a navigation event.

Reply:
Yeah, seems reasonable. Then let's leave a small comment saying that this does not trigger a navigation event.